src/Glpi/AbstractDatabase.php

```diff
@@ -124,19 +124,55 @@ abstract class AbstractDatabase
     /**
      * Execution time (can be a boolean flag to enable computation or the computation result).
      *
-     * @var boolean|string
+     * @var string
+     */
+    protected $execution_time;
+
+    /**
+     * Enable timer
+     *
+     * @var boolean
      */
-    public $execution_time = false;
+    protected $timer_enabled = false;
 
     /**
      * Constructor / Connect to the database.
      *
+     * @param array   $params Connection parameters
      * @param integer $server host number
      *
      * @return void
      */
-    public function __construct($server = null)
+    public function __construct(array $params, $server = null)
     {
+        $requireds = [
+            'driver',
+            'host',
+            'user',
+            'pass',
+            'dbname'
+        ];
+        $missing = [];
+        foreach ($requireds as $required) {
+            if (!isset($params[$required])) {
```

Comment:
If element is null, it will be considered as not set. To allow usage of `null` password:
```suggestion
            if (!array_key_exists($required, $params)) {
```